### ISSUE TITLE
DataConverter does not convert for categorical, xref, mref

### ISSUE THREAD

**Issue by dennishendriksen** [molgenis/molgenis#1658]
Data converter does not convert categorical, mref, xref values but returns the input.
Expected: exception if you try to convert categorical, xref, mref value or converters to Entity (list)

**Comment by mswertz** [comment]
what is the context of this error, i.e., what should I do to reproduce?

**Comment by dennishendriksen** [comment]
Reproduce:
Add a test case to org.molgenis.data.DataConverterTest for method "Object convert(Object source, AttributeMetaData attr)". As source supply an entity id, as attribute supply a XREF attributes.
Expected:
Return value is of type Entity.
Actual:
Return value is the input source value.

**Comment by mswertz** [comment]
I.e. this is a problem for developers, right?

**Comment by dennishendriksen** [comment]
Yes, resulting in bugs for the user that are fixed by resolving references outside of the converter.

**Comment by Mark-de-Haan** [comment]
Is the OMX converter still needed? I.e. is this bug still relevant?
